gloo/transport/ibverbs/device.h

```diff
@@ -32,6 +34,45 @@ struct attr {
   int index;
 };
 
+// Scope guard for ibverbs device list.
+class IbvDevices {
+ public:
+  IbvDevices() {
+    list_ = ibv_get_device_list(&size_);
+    if (list_ == nullptr) {
+      size_ = 0;
+    }
+  }
+
+  ~IbvDevices() {
+    if (list_ != nullptr) {
+      ibv_free_device_list(list_);
+    }
+  }
+
+  int size() {
+    return size_;
+  }
+
+  struct ibv_device*& operator[](int i) {
+    return list_[i];
+  }
+
+  // Helper function that returns the list of IB device names in sorted order
+  std::vector<std::string> getDeviceNames() {
```

Comment:
Please only add std::vector<std::string> getDeviceNames() without implementation to the header, then it can be implemented in device.cc, using the ibv_devices class.